backend/src/main/java/rocks/inspectit/gepard/agentmanager/connection/service/ConnectionService.java

```diff
@@ -20,23 +21,43 @@
 @RequiredArgsConstructor
 public class ConnectionService {
 
-  private final ConcurrentHashMap<UUID, Connection> connectionCache;
+  private final ConcurrentHashMap<String, Connection> connectionCache;
 
   private final RegexQueryService regexQueryService;
 
   /**
    * Handles a connection request from an agent.
    *
+   * @param connectionId The id for the created connection.
    * @param connectRequest The request for the new connection to be created.
    * @return Connection The response containing all saved information.
    */
-  public Connection handleConnectRequest(CreateConnectionRequest connectRequest) {
+  public Connection handleConnectRequest(
+      String connectionId, CreateConnectionRequest connectRequest) {
     Connection connection = CreateConnectionRequest.toConnection(connectRequest);
-    connectionCache.put(connection.getId(), connection);
+    connectionCache.put(connectionId, connection);
```

Comment:
connectionId in the connection is connection.agent.agentId... I really think we should not do that...

Reply:
Since we now use the agent id as path variable and key in the cache, we can actually remove the id from the agent model itself and just extract it from the create-request.

Reply:
I'm also for moving the agent model to the shared configuration repo

Reply:
That would make sense to me.